File: rentalManager/Model/Customer.cpp
#include <Model.h>


Customer::Customer(string customerID) {
   customerID = customerID;
}
void Customer::addRentalItem(string todayDate,RentalItem rentedItem) {
     rentedItems.insert(pair<string, RentalItem>(todayDate, rentedItem));
}

void Customer::removeRentalItem(RentalItem toRemove) {
    auto it = find(rentedItems.begin(), rentedItems.end(), toRemove);
    if(it != rentedItems.end())
        rentedItems.erase(it, rentedItems.end());}

map<string, RentalItem> Customer::getRentalItems() {
    return rentedItems;
}



/*
Customer Data type - Navjeeven
Name (and/or ID, customer with same name, maybe make a random number generator to make an 4 digit number)
List of rentedItem datatypes
*/

File: rentalManager/Model/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Customer {

        string customerID;
        List<RentedItem> rentedItems = new List<RentedItem>();

    double balance = 0.0;

    public double getBalance()
    {
        return this.balance;
    }

    public void generateBalance()
    {
        var balance = 0.0;
         foreach (var i in rentedItems) {
            balance += i.calculateRent();
        }

        this.balance = balance;
    }

    public Customer(string customerID) {
   this.customerID = customerID;
}
public void addRentalItem(RentedItem rentedItem) {
     rentedItems.Add(rentedItem);
}

public void removeRentalItem(RentedItem toRemove) {
    var itemToRemove = rentedItems.SingleOrDefault(x => x.getDateRented() == toRemove.getDateRented());
    if (itemToRemove != null)
        rentedItems.Remove(itemToRemove);
}

public List<RentedItem> getRentalItems() {
    return rentedItems;
}

public string getCustomerID() {
    return this.customerID;
}

}